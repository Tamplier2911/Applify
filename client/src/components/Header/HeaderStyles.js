import styled from "styled-components";

export const HeaderContainer = styled.div`
  display: grid;
  /* grid-template-columns: subgrid; */
  grid-template-columns:
    [full-start] minmax(1rem, 1fr)
    [center-start] repeat(
      10,
      [col-start] minmax(min-content, 11.7rem) [col-end]
    )
    [center-end]
    minmax(1rem, 1fr) [full-end];
  grid-area: 1 / full-start / 2 / full-end;

  height: 6rem;
  width: 100%;
  position: fixed;
  z-index: 20;
  background-color: var(--cl-header);

  align-items: center;
  justify-items: center;

  box-shadow: 0 0.2rem 0.8rem var(--cl-shadow);
`;

export const HeaderContent = styled.div`
  /* max-width: 1170px; */
  /* margin: 0 auto;  */
  // color: #e4e4e4;
  grid-column: center-start / center-end;
  grid-column: 2 / 12;
`;
