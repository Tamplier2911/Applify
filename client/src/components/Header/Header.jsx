// import "./Header.scss";
import React from "react";

// JS Rendering CSS
import { HeaderContainer, HeaderContent } from "./HeaderStyles";

const Header = () => {
  return (
    <HeaderContainer>
      <HeaderContent>header</HeaderContent>
    </HeaderContainer>
  );
};

export default Header;
