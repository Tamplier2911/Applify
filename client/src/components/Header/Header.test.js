import React from 'react';
import toJson from 'enzyme-to-json';

// component
import Header from './Header';

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from '../../tests/testUtils';

describe('<Header />', () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: 'eng',
    },
    user: {
      userObject: {
        id: '',
        name: '',
        email: '',
        photo: '',
        about: '',
        likedBlogposts: [],
      },
      errorMessage: null,
      isLogged: false,
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<Header store={store} />);
  });

  it('renders without an error', () => {
    const header = findByTestAttr(wrapper, 'header');
    expect(header.length).toBe(1);
  });

  it('renders auth block when user is logged out', () => {
    const headerLoginLink = findByTestAttr(wrapper, 'header-login-link');
    const headerSignupLink = findByTestAttr(wrapper, 'header-signup-link');
    expect(headerLoginLink.length).toBe(1);
    expect(headerSignupLink.length).toBe(1);
  });

  it('renders logout button when user is logged', () => {
    const initialStore = {
      lang: {
        languagePanelHidden: true,
        currentLanguage: 'eng',
      },
      user: {
        isLogged: true,
      },
    };
    const store = storeFactory(initialStore);
    wrapper = setShallowC(<Header store={store} />);
    const headerLogoutButton = findByTestAttr(wrapper, 'header-logout-btn');
    expect(headerLogoutButton.length).toBe(1);
  });

  it('performs logUserOutStart action on click', () => {
    const mockAction = jest.fn();
    const initialStore = {
      lang: {
        languagePanelHidden: true,
        currentLanguage: 'eng',
      },
      user: {
        isLogged: true,
      },
    };
    const store = storeFactory(initialStore);
    wrapper = setShallowC(<Header store={store} />);
    wrapper.setProps({ logUserOutStart: mockAction });
    const headerLogoutButton = findByTestAttr(wrapper, 'header-logout-btn');
    headerLogoutButton.simulate('click');
    const getClickCount = mockAction.mock.invocationCallOrder.length;
    expect(getClickCount).toBe(1);
  });

  it('renders logo link', () => {
    const logoLink = findByTestAttr(wrapper, 'header-logo-link');
    expect(logoLink.length).toBe(1);
  });

  it('renders menu button', () => {
    const initialStore = {
      lang: {
        languagePanelHidden: true,
        currentLanguage: 'eng',
      },
      user: {
        isLogged: true,
      },
      sidenav: {
        sidenavHidden: true,
      },
    };
    const store = storeFactory(initialStore);
    wrapper = setShallowC(<Header store={store} />);
    const headerMenuButton = findByTestAttr(wrapper, 'header-menu-btn');
    expect(headerMenuButton.length).toBe(1);
  });

  it('open sidenav on menu button click', () => {
    const initialStore = {
      lang: {
        languagePanelHidden: true,
        currentLanguage: 'eng',
      },
      user: {
        isLogged: true,
      },
      sidenav: {
        sidenavHidden: true, // sidenav initially hidden
      },
    };
    const store = storeFactory(initialStore);
    wrapper = setShallowC(<Header store={store} />);
    const headerMenuButton = findByTestAttr(wrapper, 'header-menu-btn');
    headerMenuButton.simulate('click');
    const {
      sidenav: { sidenavHidden },
    } = store.getState();
    expect(sidenavHidden).toBe(false); // asserting sidnenav to be open
  });

  it('matches snapshot', () => {
    const header = findByTestAttr(wrapper, 'header');
    expect(toJson(header)).toMatchSnapshot();
  });
});
