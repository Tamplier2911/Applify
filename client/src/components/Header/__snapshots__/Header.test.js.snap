// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<styled.header
  data-test="header"
  shadow={1}
>
  <styled.div>
    <styled.div>
      <Styled(Link)
        data-test="header-login-link"
        to="/authorization"
      >
        Log In
      </Styled(Link)>
       
      /
       
      <Styled(Link)
        data-test="header-signup-link"
        to="/authorization"
      >
        Sign Up
      </Styled(Link)>
    </styled.div>
    <Styled(Link)
      data-test="header-logo-link"
      to="/"
    >
      <Styled(Component) />
    </Styled(Link)>
    <Styled(Component)
      data-test="header-menu-btn"
      onClick={[Function]}
    />
  </styled.div>
</styled.header>
`;
